content/assignments/minishell/src/utils.c

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "utils.h"
+
+/**
+ * Concatenate parts of the word to obtain the command.
+ */
+char *get_word(word_t *s)
+{
+	char *string = NULL;
+	int string_length = 0;
+
+	const char *substring = NULL;
+	int substring_length = 0;
+
+	while (s != NULL) {
+		if (s->expand == true) {
+			substring = getenv(s->string);
+
+			/* Prevents strlen from failing. */
+			if (substring == NULL)
+				substring = "";
+		} else {
+			substring = s->string;
+		}
+
+		substring_length = strlen(substring);
+
+		string = realloc(string, string_length + substring_length + 1);
+		DIE(string == NULL, "Error allocating word string.");
+
+		string[string_length] = '\0';
+		strcat(string, substring);
+
+		string_length += substring_length;
+
+		s = s->next_part;
+	}
+
+	return string;
+}
+
+/**
+ * Concatenate command arguments in a NULL terminated list in order to pass
+ * them directly to execv.
+ */
+char **get_argv(simple_command_t *command, int *size)
+{
+	char **argv;
+	int argc, _argc;
```

Comment:
Why don't you reuse `argc` below? It seems pointless to declare 2 variables for the same thing.